rust/agents/relayer/src/msg/serial_submitter.rs

```diff
@@ -160,8 +167,13 @@ async fn prepare_task(
             PendingOperationResult::Success => {
                 debug!(?op, "Operation prepared");
                 metrics.ops_prepared.inc();
-                // this send will pause this task if the submitter is not ready to accept yet
-                tx_submit.send(op).await?;
+                if op.submitted() {
```

Comment:
Tbh I don't love that this is captured in an struct field as opposed to a PendingOperationResult value.

Reply:
I ended up moving this logic internally so it does not do anything differently here.